### PR TITLE
Return HTTP Bad Request 400 code when the token and uidb64 are not valid

### PR DESCRIPTION
**Pull request by KevinEtienne**
Related to #98 where a 401 was implemented

### CODE REVIEW

**Comment by Ian-Foote** [comment]
I'm still not convinced that `401` is inappropriate here.

**Comment by KevinEtienne** [comment]
@Ian-Foote @meshy @jturnbull @perry do you want to debate/review/merge? :)

File: user_management/api/tests/test_exceptions.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+from ..exceptions import InvalidExpiredToken
+
+
+class InvalidExpiredTokenTest(TestCase):
+    """Assert `InvalidExpiredToken` behaves as expected."""
+    def test_raise(self):
+        """Assert `InvalidExpiredToken` can be raised."""
+        with self.assertRaises(InvalidExpiredToken) as e:
```
Comment (Ian-Foote) [inline user_management/api/tests/test_exceptions.py]: I'd prefer `as error` or `as exception`.

File: user_management/api/views.py [hunk 2]
```diff
@@ -154,11 +152,11 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise AuthenticationFailed(detail=self.message)
+            raise exceptions.InvalidExpiredToken(detail=self.message)
 
         token = kwargs['token']
         if not default_token_generator.check_token(self.user, token):
-            raise AuthenticationFailed(detail=self.message)
+            raise exceptions.InvalidExpiredToken(detail=self.message)
```
Comment (Ian-Foote) [inline user_management/api/views.py]: The `InvalidExpiredToken` exception has the same default message as you're adding here.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## v4.2.1 (Upcoming)
```
Comment (Ian-Foote) [inline CHANGELOG.md]: I'm not sure `v4.2.1` is appropriate.

Comment (meshy) [inline CHANGELOG.md]: I think this has been bikeshedded enough. Let's please just call it `v5`, and :shipit:

Comment (KevinEtienne) [inline CHANGELOG.md]: cool

**Comment by KevinEtienne** [comment]
Updated

**Pull request merged**